### PR TITLE
IntelliJ README improvements

### PR DESCRIPTION
**Pull request by SethTisue**
I noticed a typo and some repeated text, so while fixing those, I took
the opportunity to upgrade the whole thing.

### CODE REVIEW

File: src/intellij/README.md [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
 
-Compilation withing IDEA is performed in "-Dlocker.skip=1" mode: the sources are built
-directly using the STARR compiler.
+## Requirements
+
+Use the latest IntelliJ release and install the Scala plugin from within the IDE.
+
+## Initial setup
+
+The following steps are required to use IntelliJ on Scala trunk
```
Comment (janekdb) [inline src/intellij/README.md]: Haven't seen a trunk around these parts for a while.

Comment (SethTisue) [inline src/intellij/README.md]: fixed

**Comment by SethTisue** [comment]
review by @lrytz?

**Comment by lrytz** [comment]
LGTM

**Pull request merged**